Lib/test/test_os.py

```diff
@@ -2077,6 +2077,13 @@ def test_file_link(self):
         self.assertTrue(os.path.islink(self.filelink))
         self.check_stat(self.filelink, self.filelink_target)
 
+    @unittest.skipUnless(hasattr(os, 'readlink'), 'needs os.readlink()')
```

Comment:
Needs also `os.symlink()`, isn't?

You can test `os.readlink()` without involving `os.symlink()` by calling it with non-symlinks.

Reply:
`Win32SymlinkTests` is already wrapped by `@skip_unless_symlink`, so I skipped it.

Your suggestion about using `os.readlink()` without `os.symlink()` is much better than mine, though.

Reply:
Ah, this is in Win32SymlinkTests? I have not found tests for `os.readlink()` on Posix. I think `os.readlink()` tests should be common for Posix and Windows.

Of course testing `os.readlink()` only without `os.symlink()` is not enough. But this can increase the test coverage of `os.readlink()` for the case when creating symlinks is not permitted for some reasons.

Reply:
Yes, unfortunately the test coverage of `os.readlink()` is pretty low. I'll try to improve it by creating a separate test case.